W3ChampionsStatisticService/Matches/OngoingMatchesCache.cs

```diff
@@ -38,24 +126,46 @@ public async Task<List<OnGoingMatchup>> LoadOnGoingMatches(
         int maxMmr,
         string sort)
     {
-        await UpdateCacheIfNeeded();
-
-        var matches = _values
-            .Where(m => (gameMode == GameMode.Undefined || m.GameMode == gameMode)
-                        && (gateWay == GateWay.Undefined || m.GateWay == gateWay)
-                        && (map == "Overall" || m.Map == map)
-                        && (minMmr == 0 || !m.Teams.Any(team => team.Players.Any(player => player.OldMmr < minMmr)))
-                        && (maxMmr == 3000 || !m.Teams.Any(team => team.Players.Any(player => player.OldMmr > maxMmr))));
+        var cacheKey = string.Join("|", gameMode, gateWay, map, minMmr, maxMmr, sort);
+        IEnumerable<OnGoingMatchup> matches;
+        if (_loadOngoingMatchesCache.Get(cacheKey) is CacheResult<List<OnGoingMatchup>> cachedValue)
+        {
+            if (cachedValue.IsNegativeCache)
+            {
+                return new List<OnGoingMatchup>();
+            }
 
-        if (sort == "mmrDescending")
+            matches = cachedValue.Value;
+        }
+        else
         {
-            matches = matches.OrderByDescending(m => GetMaxMmrInMatch(m));
+            await EnsureCachesInitialized();
+
+            matches = _ongoingMatchesCache.Values
+                .Where(m => (gameMode == GameMode.Undefined || m.GameMode == gameMode)
+                            && (gateWay == GateWay.Undefined || m.GateWay == gateWay)
+                            && (map == "Overall" || m.Map == map)
+                            && (minMmr == 0 || !m.Teams.Any(team => team.Players.Any(player => player.OldMmr < minMmr)))
+                            && (maxMmr == 3000 || !m.Teams.Any(team => team.Players.Any(player => player.OldMmr > maxMmr))));
+
+            matches = sort switch
+            {
+                "mmrDescending" => matches.OrderByDescending(GetMaxMmrInMatch),
```

Comment:
What about ascending? I assume they're valid if given in the API, even if they're technically the default when not given any values?